libs/frontend/feature-build/src/lib/dnd/use-drag-drop.ts

```diff
@@ -281,6 +283,12 @@ export const useDragDrop = (props: IUseDragDropProps): IUseDragDrop => {
         }
       }
     }
+
+    if (site.editor?.buildSubmenu === BuildSubmenu.New && dragSource.value?.addData) {
+      // Hide new component menu when a new component is dropped
+      setBuildSubmenu(site.editor, undefined)
+    }
```

Comment:
I eventually hide the new component menu when a new component is being dropped in the editor (either in the component tree or in another component). I think this behavior feels more natural to me, but I'm not sure if this also makes sense to you 😬.

Reply:
I'm fine either way, but I think you're right, it feels a little more natural this way. It could be annoying if you're trying to add a lot of components quickly, but maybe we can add component hotkeys for "power users" later on.